### DISCUSSION TITLE
How to update a registry record through admin token

### DISCUSSION THREAD

**Discussion by pallakartheekreddy** [Sunbird-RC/community#359] [Questions (FAQs)] [answered]
We are using Sunbird-RC to store some user info, and we observed, we need to pass an user auth token to update the user details. But we have a requirement like an admin can update some properties of all the records. Do we have any API support for this?

@maheshkumargangula @shiva-rakshith

**Comment by tejash-jl** [comment, marked as answer]
@pallakartheekreddy you need to update your schema configuration to allow users with admin role to manage the data.
https://docs.sunbirdrc.dev/use/schema-configuration#add-registry-based-on-roles

**Reply by tejash-jl** [reply]
You can use the same update API with a valid token

**Reply by pallakartheekreddy** [reply]
@tejash-jl We need to update like `"roles": ["admin"]` or `"updateRoles": ["admin"]` in schema. Can you please provide some sample?

**Reply by tejash-jl** [reply]
right now `roles` allow that user to do all CRUD operations.

**Reply by shiva-rakshith** [reply]
Hi @tejash-jl 

I have added roles to my schema. I tried updating the record using the token which has the role, But it is not updating and giving the below error:
```
User is not allowed to perform the operation on this entity
```
Schema: https://github.com/Swasth-Digital-Health-Foundation/sunbird-rc-core/blob/c51e942a7fe66bcc50a9de981d9d71ba0cdd042f/java/registry/src/main/resources/public/_schemas/Organisation.json#L176

Can you please help?
